src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,43 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int BASE_CAPACITY = 0;
+    private KeyValue<K, V>[] keyValueArray;
+
+    public StorageImpl() {
+        keyValueArray = new KeyValue[BASE_CAPACITY];
```

Comment:
You should cast your array 
```suggestion
        keyValueArray = (KeyValue<K, V>[]) new KeyValue[BASE_CAPACITY];
```